maven-archetype-plugin/src/main/java/org/apache/maven/archetype/ui/generation/DefaultArchetypeGenerationConfigurator.java

```
@@ -368,71 +351,113 @@ private void restoreCommandLineProperties( ArchetypeConfiguration archetypeConfi
         }
 
-        private int comparePropertyName( String left, String right )
+        private Map<String, Set<String>> computePropertyReferences()
         {
-            if ( "groupId".equals( left ) )
-            {
-                return -1;
-            }
-            if ( "groupId".equals( right ) )
-            {
-                return 1;
-            }
-            if ( "artifactId".equals( left ) )
-            {
-                return -1;
-            }
-            if ( "artifactId".equals( right ) )
+            Map<String, Set<String>> result = new HashMap<>();
+
+            List<String> requiredProperties = archetypeConfiguration.getRequiredProperties();
+
+            for ( String propertyName : requiredProperties )
             {
-                return 1;
+                final Set<String> referencedPropertyNames = new LinkedHashSet<>();
+
+                String defaultValue = archetypeConfiguration.getDefaultValue( propertyName );
+                if ( StringUtils.contains( defaultValue, "${" ) )
+                {
+                    try
+                    {
+                        final boolean dumpNamespace = false;
+                        SimpleNode node = RuntimeSingleton.parse(
+                                        new StringReader( defaultValue ), propertyName + ".default", dumpNamespace );
+                        node.jjtAccept( new BaseVisitor()
+                        {
+                            @SuppressWarnings( "unchecked" )
+                            @Override
+                            public Object visit( ASTReference node, Object data )
+                            {
+                                ( ( Set<String> ) data ).add( node.getFirstToken().next.image );
+                                return super.visit( node, data );
+                            }
+                        }, referencedPropertyNames );
+                    }
+                    catch ( ParseException e )
```

if an empty catch block is really OK here, please add a comment explaining why